mobile/ios/ios-core.ts

```diff
@@ -980,44 +984,104 @@ export class PlistService implements Mobile.IiOSDeviceSocket {
 	}
 }
 
-export class GDBServer implements Mobile.IGDBServer {
-	private socket: Mobile.IiOSDeviceSocket  = null;
+function getCharacterCodePoint(ch: string) {
+	assert.equal(ch.length, 1);
 
-	constructor(private service: number,
-		private $injector: IInjector) {
+	var code = ch.charCodeAt(0);
+	assert.ok(!(0xD800 <= code && code <= 0xffff));
 
-		if(hostInfo.isWindows()) {
-			this.socket = this.$injector.resolve(WinSocket, {service: this.service, format: 0});
-		} else if(hostInfo.isDarwin()) {
-			this.socket = this.$injector.resolve(PosixSocket, {service: this.service, format: CoreTypes.kCFPropertyListXMLFormat_v1_0});
+	return code;
+}
+
+util.inherits(GDBStandardOutputAdapter, stream.Transform);
+function GDBStandardOutputAdapter(opts?: stream.TransformOptions) {
```

Comment:
We just verified with @fatme, and it seems that indeed, this can be written as a class. You need to use this code:

```
class GDBStandardOutputAdapter extends stream.Transform {
	constructor(opts?: stream.TransformOptions) {
		super(opts);
		util.inherits(GDBStandardOutputAdapter, stream.Transform);

		this.utf8StringDecoder = new string_decoder.StringDecoder("utf8");
	}

	_transform(packet: any, encoding: string, done: Function): void {
		// your code for _transform here
                // note: _transform could be called with either Buffer or string and you need to either add a check or convert to the type you expect
	}

```